coalib/core/DependencyTracker.py

```diff
@@ -97,6 +97,65 @@ def get_dependencies(self, dependant):
             for dependency, dependants in self._dependency_dict.items()
             if dependant in dependants)
 
+    def get_all_dependants(self, dependency):
+        """
+        Returns a set of all dependants of the given dependency, even
+        indirectly related ones.
+
+        >>> tracker = DependencyTracker()
+        >>> tracker.add(0, 1)
+        >>> tracker.add(1, 2)
+        >>> tracker.get_all_dependants(0)
+        {1, 2}
+
+        :param dependency:
+            The dependency to get all dependants for.
+        :return:
+            A set of dependants.
+        """
+        dependants = set()
+
+        def append_to_dependants(prev, nxt):
+            dependants.add(nxt)
+
+        traverse_graph(
+            [dependency],
+            lambda node: self._dependency_dict.get(node, frozenset()),
```

Comment:
no that's the only difference :+1:
Actually I thought this gives a slight performance improvement as frozenset doesn't require an extra allocation, though the same would apply to a normal set, as it is reserved once before `traverse_graph` is called for the passed lambda. If you want I change it to set, it doesn't really matter^^